django/db/models/lookups.py

```diff
@@ -406,6 +406,14 @@ def split_parameter_list_as_sql(self, compiler, connection):
         return ''.join(in_clause_elements), params
 
 
+@Field.register_lookup
+class NotIn(In):
```

Comment:
This won't work as is because `In.process_rhs` optimizes empty result sets which breaks vacuous truth in this case. In other words `.filter(field__notin=[])` won't return *any* results when it should return *all* results.

Reply:
Thanks @charettes - I'll work on this a bit more. I appreciate you taking the time to look through this.